doc/rbd/iscsi-initiator-esx.rst

```diff
@@ -6,32 +6,100 @@ iSCSI Initiator for VMware ESX
+   required by target" and enter the CHAP credentials used in the gwcli
+   auth command or ansible client_connections credentials variable.
+
+   The Mutual CHAP authentication section should have "Do not use CHAP"
+   selected.
+
+   Warning: There is a bug in the web client where the requested CHAP
+   settings are not always used initially. On the iSCSI gateway kernel
+   logs you will see the error:
+
+    ::
+
+       > kernel: CHAP user or password not set for Initiator ACL
+       > kernel: Security negotiation failed.
+       > kernel: iSCSI Login negotiation failed.
+
+   To workaround this set the CHAP settings with the esxcli command. Here
+   authname is the username and secret is the password used in previous
+   examples:
+
+   ::
+
+       > esxcli iscsi adapter auth chap set --direction=uni --authname=myiscsiusername --secret=myiscsipassword --level=discouraged -A vmhba64
+
+#. Configure iSCSI Settings
+
+   .. image:: ../images/esx_iscsi_recov_timeout.png
+      :align: center
+
+   Expand Advanced settings and set the "RecoveryTimeout" to 25.
 
-#. In the General tab click the "Advanced" button and in the "Advanced Settings"
-   set RecoveryTimeout to 25.
+#. Set the discovery address
 
-#. If CHAP was setup on the iSCSI gateway, in the General tab click the "CHAP..."
-   button. If CHAP is not being used, skip to step 4.
+   .. image:: ../images/esx_config_iscsi_main.png
+      :align: center
 
-#. On the CHAP Credentials windows, select “Do not use CHAP unless required by target”,
-   and enter the "Name" and "Secret" values used on the initial setup for the iSCSI
-   gateway, then click on the "OK" button.
+   In the Dynamic targets section, click "Add dynamic target" and under
+   Addresses add one of the gateway IP addresses added during the iSCSI
+   gateway setup stage in the gwcli section or an IP set in the ansible
+   gateway_ip_list variable. Only one address needs to be added as the gateways
+   have been setup so all the iSCSI portals are returned during discovery.
 
-#. On the Dynamic Discovery tab, click the "Add..." button, and enter the IP address
-   and port of one of the iSCSI target portals. Click on the "OK" button.
+   Finally, click the "Save configuration" button. In the Devices tab, you
+   should see the RBD image.
 
-#. Close the iSCSI Initiator Properties window. A prompt will ask to rescan the
-   iSCSI software adapter. Select Yes.
+   The LUN should be automatically configured and using the ALUA SATP and
+   MRU PSP. Other SATPs and PSPs must not be used. This can be verified with
+   the esxcli command:
 
-#. In the Details pane, the LUN on the iSCSI target will be displayed. Right click
-   on a device and select "Manage Paths".
+   ::
 
-#. On the Manage Paths window, select “Most Recently Used (VMware)” for the policy
-   path selection. Close and repeat for the other disks.
+       > esxcli storage nmp path list -d eui.your_device's_id
```

Comment:
Nit: the apostrophe creates a partial quote effect in the rendered docs

Reply:
Fixed.